Cabal/Distribution/C2Hs.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+-- Based on https://github.com/gtk2hs/gtk2hs/blob/master/tools/src/Gtk2HsSetup.hs#L414
+module Distribution.C2Hs ( reorderC2Hs ) where
+
+import Prelude()
+import Distribution.Compat.Prelude
+import qualified Data.Map as M
+import qualified Data.Set as S
+import Distribution.C2Hs.Lexer
+import Distribution.ModuleName (ModuleName, components)
+import Distribution.Parsec (simpleParsec)
+import Distribution.Simple.Utils (dieNoVerbosity, findFileWithExtension, withUTF8FileContents)
+import System.FilePath (joinPath)
+
+reorderC2Hs :: [FilePath] -> [ModuleName] -> IO [ModuleName]
```

Comment:
This need a proper comment, what it does, and why.

Reply:
Will do.